rust/src/lib/ovsdb/db.rs

```diff
@@ -396,26 +315,29 @@ pub(crate) fn parse_uuid_array(v: &[Value]) -> Vec<String> {
     ret
 }
 
-#[derive(Debug, Clone, Default, PartialEq, Eq)]
-pub(crate) struct OvsDbUpdate {
-    pub(crate) table: String,
-    pub(crate) conditions: Vec<OvsDbCondition>,
-    pub(crate) row: HashMap<String, Value>,
-}
-
-impl OvsDbUpdate {
-    fn to_value(&self) -> Value {
-        let mut ret = Map::new();
-        ret.insert("op".to_string(), Value::String("update".to_string()));
-        ret.insert("table".to_string(), Value::String(self.table.clone()));
-        let condition_values: Vec<Value> =
-            self.conditions.iter().map(|c| c.to_value()).collect();
-        ret.insert("where".to_string(), Value::Array(condition_values));
-        let mut row_map = Map::new();
-        for (k, v) in self.row.iter() {
-            row_map.insert(k.to_string(), v.clone());
+fn check_transact_error(reply: Value) -> Result<Value, NmstateError> {
+    if let Some(trans_replies) = reply.as_array() {
+        for trans_reply in trans_replies {
+            if let Some(error_type) = trans_reply
+                .as_object()
+                .and_then(|r| r.get("error"))
+                .and_then(|e| e.as_str())
+            {
+                let error_detail = trans_reply
+                    .as_object()
+                    .and_then(|r| r.get("details"))
+                    .and_then(|d| d.as_str())
+                    .unwrap_or("");
```

Comment:
Maybe put here a generic error description instead of an empty string. "unknown error"?

Reply:
Changed to `.unwrap_or("unknown error");`. Thanks.